app/src/main/java/com/kamron/pogoiv/ScreenGrabber.java

```diff
@@ -144,4 +114,55 @@ private Bitmap getBitmap(ByteBuffer buffer, int pixelStride, int rowPadding) {
 
         return bmp;
     }
+
+    public @Nullable @ColorInt int[] grabPixels(Point[] points) {
+        Image image = null;
+        try {
+            //Note: imageReader shouldn't be null, but apparently sometimes is.
+            //Let's allow this to still happen.
+            image = imageReader.acquireLatestImage();
+        } catch (Exception exception) {
+            Timber.e("Error thrown in grabPixels() - acquireLatestImage()");
+            Timber.e(exception);
+        }
+
+        if (image == null) {
+            return null;
+        }
+
+        Rect imageBounds = new Rect(0, 0, image.getWidth(), image.getHeight());
+        @ColorInt int[] pixels = new int[points.length];
+
+        Image.Plane imgPlane = image.getPlanes()[0];
+        int pixelStride = imgPlane.getPixelStride();
+        int rowStride = imgPlane.getRowStride();
+        ByteBuffer buffer = imgPlane.getBuffer();
+
+        for (int i = 0; i < points.length; i++) {
+            Point p = points[i];
+            if (imageBounds.contains(p.x, p.y)) {
+                pixels[i] = getPixel(buffer, p, pixelStride, rowStride);
```

Comment:
> Also, taking a closer look at the Bitmap docs and grabScreen() I think a simpler code with the same benefits (the main benefit of this change is to avoid copyPixelsFromBuffer) would be to use this constructor when creating the bitmap in grabScreen?

That constructor takes a int array in ARGB format. Our `Image` returns a ByteBuffer in RGBA format, from which we *might* be able to get a `byte[]` array (again in RGBA), not an `int[]` array (in ARGB). As discussed [below](https://github.com/farkam135/GoIV/pull/396#discussion_r78292182), I don't see a way to match the formats. And I don't think a `byte[]` can be cast to an `int[]` without JNI somewhere.

I've looked into `BitmapFactory` multiple times, but according to its docs (and to the available options) doesn't seem to be addressed at our use case but at files, and StackOverflow seems to agree (http://stackoverflow.com/q/26673127/53974; I wouldn't trust it if I hadn't checked the docs).

I'm not sure why the APIs are designed to *require* a conversion, but this conflict has enough precedent to give rise to a confusing Wikipedia page (which I don't really trust):
https://en.wikipedia.org/wiki/RGBA_color_space#Representation.